<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get Started with SCIntRuler • SCIntRuler</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Get Started with SCIntRuler">
<meta property="og:description" content="Guiding the integration of multiple single-cell RNA-seq datasets with a novel statistical metric.
">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SCIntRuler</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Getting Started
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Computational Resources</li>
    <li>
      <a href="../articles/SCIntRuler.html">SCIntRuler Package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/yuelyu21/SCIntRuler">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get Started with SCIntRuler</h1>
                        <h4 data-toc-skip class="author"><a href="https://orcid.org/0000-0002-8912-6624" class="external-link">Yue
Lyu
<img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_24x24.png" width="16" height="16"></a></h4>
                        
      
      <small class="dont-index">Source: <a href="https://github.com/yuelyu21/SCIntRuler/blob/HEAD/vignettes/SCIntRuler.Rmd"><code>vignettes/SCIntRuler.Rmd</code></a></small>
      <div class="hidden name"><code>SCIntRuler.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introducrtion">Introducrtion<a class="anchor" aria-label="anchor" href="#introducrtion"></a>
</h2>
<p>The accumulation of single-cell RNA-seq (scRNA-seq) studies
highlights the potential benefits of integrating multiple data sets. By
augmenting sample sizes and enhancing analytical robustness, integration
can lead to more insightful biological conclusions. However, challenges
arise due to the inherent diversity and batch discrepancies within and
across studies. <strong>SCIntRuler</strong>, a novel R package,
addresses these challenges by guiding the integration of multiple
scRNA-seq data sets. <a href="https://github.com/yuelyu21/SCIntRuler">SCIntRuler</a> is an R
package developed for single-cell RNA-seq analysis. It was designed
using the <a href="https://satijalab.org/seurat/" class="external-link">Seurat</a> framework,
and offers existing and novel single-cell analytic work flows.</p>
<p>Integrating scRNA-seq data sets can be complex due to various
factors, including batch effects and sample diversity. Key decisions –
whether to integrate data sets, which method to choose for integration,
and how to best handle inherent data discrepancies – are crucial.
<code>SCIntRuler</code> offers a statistical metric to aid in these
decisions, ensuring more robust and accurate analyses.</p>
<ul>
<li>
<strong>Informed Decision Making</strong>: Helps researchers decide
on the necessity of data integration and the most suitable method.</li>
<li>
<strong>Flexibility</strong>: Suitable for various scenarios,
accommodating different levels of data heterogeneity.</li>
<li>
<strong>Robustness</strong>: Enhances analytical robustness in joint
analyses of merged or integrated scRNA-seq data sets.</li>
<li>
<strong>User-Friendly</strong>: Streamlines decision-making
processes, simplifying the complexities involved in scRNA-seq data
integration.</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">1. Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install the package, you need to install the
<code>batchelor</code> and <code>MatrixGenerics</code> package from
Bioconductor.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Check if BioManager is installed, install if not</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check if 'batchelor' is installed, install if not</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"batchelor"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"batchelor"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check if 'MatrixGenerics' is installed, install if not</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"MatrixGenerics"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MatrixGenerics"</span><span class="op">)</span></span></code></pre></div>
<p>The <code>SCIntRuler</code> can be installed by the following
commands, the source code can be found at <a href="https://github.com/yuelyu21/SCIntRuler">GitHub</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SCIntRuler"</span><span class="op">)</span> </span></code></pre></div>
<p>After the installation, the package can be loaded with</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/yuelyu21/SCIntRuler">SCIntRuler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="explore-with-an-example-data">2. Explore with an example data<a class="anchor" aria-label="anchor" href="#explore-with-an-example-data"></a>
</h2>
<p>Let’s start with an example data. We conducted a series of simulation
studies to assess the efficacy of <code>SCIntRuler</code> in guiding the
integration selection under different scenarios with varying degrees of
shared information among data sets. We generated the simulation data
based on a real <code>Peripheral Blood Mononuclear Cells (PBMC)</code>
scRNA-seq dataset.</p>
<div class="section level3">
<h3 id="overview-of-the-data">Overview of the data<a class="anchor" aria-label="anchor" href="#overview-of-the-data"></a>
</h3>
<p>This dataset is a subset of what we used in our <strong>Simulation
2</strong>, where we have three studies. In each study, we randomly drew
different numbers of CD4 T helper cells, B cells, CD14 monocytes, and
CD56 NK cells to mimic four real-world scenarios with three data sources
<strong>Simulation 2</strong> introduces a moderate overlap, with 20.3%
cells sharing the same cell type identity. There are 2000 B cells and
400 CD4T cells in the first study, 700 CD14Mono cells and 400 CD4T cells
in the second study, 2000 CD56NK cells and 400 CD4T cells in the third
study. This data is already in <code>Seurat</code> format and can be
found under <code>/data</code>. There are 32738 genes and 5900 cells in
simulation 2. Here, we subset 800 cells with 3000 genes.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sim_data_sce"</span>, package <span class="op">=</span> <span class="st">"SCIntRuler"</span><span class="op">)</span></span>
<span><span class="va">sim_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/as.Seurat.html" class="external-link">as.Seurat</a></span><span class="op">(</span><span class="va">sim_data_sce</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sim_data</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;                     orig.ident nCount_RNA nFeature_RNA CellType  Study</span></span>
<span><span class="co">#&gt; AATTACGAATCGGT-1 SeuratProject       1035          365    Bcell Study1</span></span>
<span><span class="co">#&gt; AGAGCGGAGTCCTC-1 SeuratProject       1157          448    Bcell Study1</span></span>
<span><span class="co">#&gt; TTCTTACTGGTACT-1 SeuratProject       2824          884    Bcell Study1</span></span>
<span><span class="co">#&gt; TGACGCCTACACCA-1 SeuratProject       1801          644    Bcell Study1</span></span>
<span><span class="co">#&gt; ATTTGCACCTATGG-1 SeuratProject       2501          749    Bcell Study1</span></span>
<span><span class="co">#&gt; AGAACAGATGGAGG-1 SeuratProject       1113          391    Bcell Study1</span></span>
<span><span class="co">#&gt;                  RNA_snn_res.0.5 seurat_clusters ident</span></span>
<span><span class="co">#&gt; AATTACGAATCGGT-1               0               0     0</span></span>
<span><span class="co">#&gt; AGAGCGGAGTCCTC-1               0               0     0</span></span>
<span><span class="co">#&gt; TTCTTACTGGTACT-1               0               0     0</span></span>
<span><span class="co">#&gt; TGACGCCTACACCA-1               0               0     0</span></span>
<span><span class="co">#&gt; ATTTGCACCTATGG-1               0               0     0</span></span>
<span><span class="co">#&gt; AGAACAGATGGAGG-1               0               0     0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-pre-process-and-visulization-with-seurat">Data pre-process and visulization with Seurat<a class="anchor" aria-label="anchor" href="#data-pre-process-and-visulization-with-seurat"></a>
</h3>
<p>Followed by the tutorial of <a href="https://satijalab.org/seurat/articles/pbmc3k_tutorial" class="external-link">Seurat</a>,
we first pre-processed the data by the functions
<code>NormalizeData</code>, <code>FindVariableFeature</code>,
<code>ScaleData</code>, <code>RunPCA</code>, <code>FindNeighbors</code>,
<code>FinsClusters</code> and <code>RunUMAP</code> from
<code>Seurat</code> and then draw the UMAP by using <code>DimPlot</code>
stratified by <strong>Study</strong> and <strong>Cell Type</strong>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Normalize the data</span></span>
<span><span class="va">sim_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">sim_data</span><span class="op">)</span></span>
<span><span class="co"># Identify highly variable features</span></span>
<span><span class="va">sim_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">sim_data</span>, selection.method <span class="op">=</span> <span class="st">"vst"</span>, nfeatures <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span></span>
<span><span class="co"># Scale the data</span></span>
<span><span class="va">all.genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">sim_data</span><span class="op">)</span></span>
<span><span class="va">sim_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">sim_data</span>, features <span class="op">=</span> <span class="va">all.genes</span><span class="op">)</span></span>
<span><span class="co"># Perform linear dimensional reduction</span></span>
<span><span class="va">sim_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">sim_data</span>, features <span class="op">=</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/VariableFeatures.html" class="external-link">VariableFeatures</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">sim_data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># Cluster the cells</span></span>
<span><span class="va">sim_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindNeighbors.html" class="external-link">FindNeighbors</a></span><span class="op">(</span><span class="va">sim_data</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="va">sim_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindClusters.html" class="external-link">FindClusters</a></span><span class="op">(</span><span class="va">sim_data</span>, resolution <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 800</span></span>
<span><span class="co">#&gt; Number of edges: 46490</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.7653</span></span>
<span><span class="co">#&gt; Number of communities: 4</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sim_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">sim_data</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="umap-separated-by-study">UMAP separated by Study<a class="anchor" aria-label="anchor" href="#umap-separated-by-study"></a>
</h4>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">sim_data</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">FALSE</span>, pt.size <span class="op">=</span> <span class="fl">.5</span>, group.by <span class="op">=</span> <span class="st">"Study"</span>, repel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p1</span></span></code></pre></div>
<p><img src="SCIntRuler_files/figure-html/UMAP%20-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="umap-separated-by-cell-type">UMAP separated by cell type<a class="anchor" aria-label="anchor" href="#umap-separated-by-cell-type"></a>
</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">sim_data</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, label <span class="op">=</span> <span class="cn">TRUE</span>, pt.size <span class="op">=</span> <span class="fl">.8</span>, group.by <span class="op">=</span> <span class="st">"CellType"</span>, repel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">p2</span></span></code></pre></div>
<p><img src="SCIntRuler_files/figure-html/UMAP2%20-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="try-different-data-integration-methods">Try different data integration methods<a class="anchor" aria-label="anchor" href="#try-different-data-integration-methods"></a>
</h3>
<p>To further illustrate which integration method is more suitable under
different settings, we visualize the data without integration (simply
merging the single cell objects) and after applying three popular data
integration methods: CCA, Harmony and Scanorama. The UMAP visualizations
across the simulations indicate that the choice of integration method
significantly impacts the resulting data integration.</p>
<div class="section level4">
<h4 id="run-seurat-cca">Run Seurat CCA<a class="anchor" aria-label="anchor" href="#run-seurat-cca"></a>
</h4>
<p>Seurat CCA can be directly applied by functions
<code>FindIntegrationAnchors</code> and <code>IntegrateData</code> in
<code>Seurat</code> package. For more information, please see<a href="https://satijalab.org/seurat/articles/integration_introduction.html" class="external-link">Tutorial
of SeuratCCA</a>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### CCA</span></span>
<span><span class="va">sim.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject</a></span><span class="op">(</span><span class="va">sim_data</span>, split.by <span class="op">=</span> <span class="st">"Study"</span><span class="op">)</span></span>
<span><span class="va">sim.anchors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindIntegrationAnchors.html" class="external-link">FindIntegrationAnchors</a></span><span class="op">(</span>object.list <span class="op">=</span> <span class="va">sim.list</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, reduction <span class="op">=</span> <span class="st">"cca"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sim.int</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/IntegrateData.html" class="external-link">IntegrateData</a></span><span class="op">(</span>anchorset <span class="op">=</span> <span class="va">sim.anchors</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, new.assay.name <span class="op">=</span> <span class="st">"CCA"</span><span class="op">)</span></span>
<span><span class="co"># run standard analysis workflow</span></span>
<span><span class="va">sim.int</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">sim.int</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">sim.int</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">sim.int</span>, npcs <span class="op">=</span> <span class="fl">30</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">sim.int</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">sim.int</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, reduction.name <span class="op">=</span> <span class="st">"umap_cca"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="run-harmony">Run Harmony<a class="anchor" aria-label="anchor" href="#run-harmony"></a>
</h4>
<p>Harmony is an algorithm for performing integration of single cell
genomics data sets. To run harmony, we need to install
<code>harmony</code> package. For more information, please see <a href="https://portals.broadinstitute.org/harmony/articles/quickstart.html" class="external-link">Quick
start to Harmony</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Harmony</span></span>
<span><span class="co"># install.packages("harmony")</span></span>
<span></span>
<span><span class="va">sim.harmony</span> <span class="op">&lt;-</span> <span class="fu">harmony</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/harmony/man/RunHarmony.html" class="external-link">RunHarmony</a></span><span class="op">(</span><span class="va">sim_data</span>, group.by.vars <span class="op">=</span> <span class="st">"Study"</span>, reduction.use <span class="op">=</span> <span class="st">"pca"</span>,</span>
<span>                                   <span class="co">#dims.use = 1:20, assay.use = "RNA"</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">sim.int</span><span class="op">[[</span><span class="st">"harmony"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">sim.harmony</span><span class="op">[[</span><span class="st">"harmony"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">sim.int</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">sim.int</span>, dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">20</span>, reduction <span class="op">=</span> <span class="st">"harmony"</span>, reduction.name <span class="op">=</span> <span class="st">"umap_harmony"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="umap-figures-for-all-integrated-data">UMAP figures for all integrated data<a class="anchor" aria-label="anchor" href="#umap-figures-for-all-integrated-data"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">sim_data</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"Study"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>        <span class="co"># axis.line.y = element_line( size = 2, linetype = "solid"),</span></span>
<span>        <span class="co"># axis.line.x = element_line( size = 2, linetype = "solid"),</span></span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p6</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">sim.int</span>, reduction <span class="op">=</span> <span class="st">"umap_cca"</span>, group.by <span class="op">=</span> <span class="st">"Study"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>        <span class="co"># axis.line.y = element_line( size = 2, linetype = "solid"),</span></span>
<span>        <span class="co"># axis.line.x = element_line( size = 2, linetype = "solid"),</span></span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p7</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">sim.int</span>, reduction <span class="op">=</span> <span class="st">"umap_harmony"</span>, group.by <span class="op">=</span> <span class="st">"Study"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>        <span class="co"># axis.line.y = element_line( size = 2, linetype = "solid"),</span></span>
<span>        <span class="co"># axis.line.x = element_line( size = 2, linetype = "solid"),</span></span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span> color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span>,</span>
<span>        axis.title.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">"black"</span>, size<span class="op">=</span><span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">leg</span> <span class="op">&lt;-</span> <span class="fu">cowplot</span><span class="fu">::</span><span class="fu"><a href="https://wilkelab.org/cowplot/reference/get_legend.html" class="external-link">get_legend</a></span><span class="op">(</span><span class="va">p5</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in get_plot_component(plot, "guide-box"): Multiple components found;</span></span>
<span><span class="co">#&gt; returning the first one. To return all, use `return_all = TRUE`.</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span><span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">arrangeGrob</a></span><span class="op">(</span><span class="va">p5</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Unintegrated"</span><span class="op">)</span>, </span>
<span>                                               <span class="va">p6</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Seurat CCA"</span><span class="op">)</span> , </span>
<span>                                               <span class="va">p7</span> <span class="op">+</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SeuratTheme.html" class="external-link">NoLegend</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Harmony"</span><span class="op">)</span>,</span>
<span>                                               <span class="co">#p8 + NoLegend() + ggtitle("Scanorama"), </span></span>
<span>                                               nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>    <span class="va">leg</span>, ncol <span class="op">=</span> <span class="fl">2</span>, widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">20</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="SCIntRuler_files/figure-html/integration-1.png" width="1344"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="applying-scintruler-to-example-data">3. Applying SCIntRuler to example data<a class="anchor" aria-label="anchor" href="#applying-scintruler-to-example-data"></a>
</h2>
<p>We first split the <code>sim_data</code> by <code>Study</code> and
then run <code>GetCluster</code> and <code>NormData</code> to get
Louvain clusters and normalized count matrix for each study.
Furthermore, to perform the permutation test of relative-between cluster
distance, <code>FindNNDist</code> can be then applied. We also have
another function <code>FindNNDistC</code> that based on
<code>Rcpp</code> and <code>C++</code> for faster matrix
calculation.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">sim.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject</a></span><span class="op">(</span><span class="va">sim_data</span>, split.by <span class="op">=</span> <span class="st">"Study"</span><span class="op">)</span></span>
<span><span class="va">fullcluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetCluster.html">GetCluster</a></span><span class="op">(</span><span class="va">sim.list</span>,<span class="fl">50</span>,<span class="fl">200</span><span class="op">)</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 336</span></span>
<span><span class="co">#&gt; Number of edges: 21610</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.5099</span></span>
<span><span class="co">#&gt; Number of communities: 2</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 136</span></span>
<span><span class="co">#&gt; Number of edges: 4813</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.7201</span></span>
<span><span class="co">#&gt; Number of communities: 2</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span>
<span><span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Number of nodes: 328</span></span>
<span><span class="co">#&gt; Number of edges: 22343</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Running Louvain algorithm...</span></span>
<span><span class="co">#&gt; Maximum modularity in 10 random starts: 0.5141</span></span>
<span><span class="co">#&gt; Number of communities: 2</span></span>
<span><span class="co">#&gt; Elapsed time: 0 seconds</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">normCount</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NormData.html">NormData</a></span><span class="op">(</span><span class="va">sim.list</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#distmat &lt;- FindNNDist(fullcluster, normCount, 20)</span></span>
<span><span class="va">distmat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FindNNDistC.html">FindNNDistC</a></span><span class="op">(</span><span class="va">fullcluster</span>, <span class="va">normCount</span>, <span class="fl">20</span><span class="op">)</span></span>
<span><span class="co">#&gt;   |                                                                    |                                                            |   0%  |                                                                    |====================                                        |  33%  |                                                                    |========================================                    |  67%  |                                                                    |============================================================| 100%</span></span></code></pre></div>
<div class="section level3">
<h3 id="calculate-scintruler-and-and-an-associated-visualization">Calculate SCIntRuler and and an associated visualization<a class="anchor" aria-label="anchor" href="#calculate-scintruler-and-and-an-associated-visualization"></a>
</h3>
<p>In this example data, we got a SCIntRuler score of 0.57, there is a
noticeable but not large overlap of cell types across the data sets,
showing a moderate level of shared information. The integration is
essential to adjust for these effects and align the shared cell
populations, ensuring that the integrated dataset accurately reflects
the biological composition. Thus, the methods which can offer a balance
between correcting for batch effects and maintaining biological
variation would be the best. The UMAP visualizations across the
simulations indicate that the choice of integration method significantly
impacts the resulting data integration.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">testres</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PermTest.html">PermTest</a></span><span class="op">(</span><span class="va">fullcluster</span>,<span class="va">distmat</span>,<span class="fl">15</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/PlotSCIR.html">PlotSCIR</a></span><span class="op">(</span><span class="va">fullcluster</span>,<span class="va">sim.list</span>,<span class="va">testres</span>,<span class="st">""</span><span class="op">)</span></span></code></pre></div>
<p><img src="SCIntRuler_files/figure-html/step%202-1.png" width="700"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">sim_result</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">fullcluster</span>,<span class="va">normCount</span>,<span class="va">distmat</span>,<span class="va">testres</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] ggplot2_3.5.1      dplyr_1.1.4        Seurat_5.1.0       SeuratObject_5.0.2</span></span>
<span><span class="co">#&gt; [5] sp_2.1-4           SCIntRuler_0.99.3  BiocStyle_2.32.1  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RcppAnnoy_0.0.22            splines_4.4.1              </span></span>
<span><span class="co">#&gt;   [3] later_1.3.2                 batchelor_1.20.0           </span></span>
<span><span class="co">#&gt;   [5] tibble_3.2.1                polyclip_1.10-6            </span></span>
<span><span class="co">#&gt;   [7] fastDummies_1.7.3           lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;   [9] globals_0.16.3              lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [11] MASS_7.3-60.2               magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;  [13] plotly_4.10.4               sass_0.4.9                 </span></span>
<span><span class="co">#&gt;  [15] rmarkdown_2.27              jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [17] yaml_2.3.8                  httpuv_1.6.15              </span></span>
<span><span class="co">#&gt;  [19] sctransform_0.4.1           spam_2.10-0                </span></span>
<span><span class="co">#&gt;  [21] spatstat.sparse_3.1-0       reticulate_1.38.0          </span></span>
<span><span class="co">#&gt;  [23] cowplot_1.1.3               pbapply_1.7-2              </span></span>
<span><span class="co">#&gt;  [25] RColorBrewer_1.1-3          ResidualMatrix_1.14.1      </span></span>
<span><span class="co">#&gt;  [27] multcomp_1.4-25             abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [29] zlibbioc_1.50.0             Rtsne_0.17                 </span></span>
<span><span class="co">#&gt;  [31] GenomicRanges_1.56.1        purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [33] BiocGenerics_0.50.0         TH.data_1.1-2              </span></span>
<span><span class="co">#&gt;  [35] sandwich_3.1-0              GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">#&gt;  [37] IRanges_2.38.0              S4Vectors_0.42.0           </span></span>
<span><span class="co">#&gt;  [39] ggrepel_0.9.5               irlba_2.3.5.1              </span></span>
<span><span class="co">#&gt;  [41] listenv_0.9.1               spatstat.utils_3.0-5       </span></span>
<span><span class="co">#&gt;  [43] goftest_1.2-3               RSpectra_0.16-1            </span></span>
<span><span class="co">#&gt;  [45] spatstat.random_3.2-3       fitdistrplus_1.1-11        </span></span>
<span><span class="co">#&gt;  [47] parallelly_1.37.1           DelayedMatrixStats_1.26.0  </span></span>
<span><span class="co">#&gt;  [49] pkgdown_2.0.9               coin_1.4-3                 </span></span>
<span><span class="co">#&gt;  [51] leiden_0.4.3.1              codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [53] DelayedArray_0.30.1         scuttle_1.14.0             </span></span>
<span><span class="co">#&gt;  [55] tidyselect_1.2.1            UCSC.utils_1.0.0           </span></span>
<span><span class="co">#&gt;  [57] farver_2.1.2                ScaledMatrix_1.12.0        </span></span>
<span><span class="co">#&gt;  [59] matrixStats_1.3.0           stats4_4.4.1               </span></span>
<span><span class="co">#&gt;  [61] spatstat.explore_3.2-7      jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt;  [63] BiocNeighbors_1.22.0        progressr_0.14.0           </span></span>
<span><span class="co">#&gt;  [65] ggridges_0.5.6              survival_3.6-4             </span></span>
<span><span class="co">#&gt;  [67] systemfonts_1.1.0           tools_4.4.1                </span></span>
<span><span class="co">#&gt;  [69] ragg_1.3.2                  ica_1.0-3                  </span></span>
<span><span class="co">#&gt;  [71] Rcpp_1.0.12                 glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [73] gridExtra_2.3               SparseArray_1.4.8          </span></span>
<span><span class="co">#&gt;  [75] xfun_0.45                   MatrixGenerics_1.16.0      </span></span>
<span><span class="co">#&gt;  [77] GenomeInfoDb_1.40.1         withr_3.0.0                </span></span>
<span><span class="co">#&gt;  [79] BiocManager_1.30.23         fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [81] fansi_1.0.6                 rsvd_1.0.5                 </span></span>
<span><span class="co">#&gt;  [83] digest_0.6.36               R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [85] mime_0.12                   textshaping_0.4.0          </span></span>
<span><span class="co">#&gt;  [87] colorspace_2.1-0            scattermore_1.2            </span></span>
<span><span class="co">#&gt;  [89] tensor_1.5                  spatstat.data_3.1-2        </span></span>
<span><span class="co">#&gt;  [91] RhpcBLASctl_0.23-42         utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [93] tidyr_1.3.1                 generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [95] data.table_1.15.4           httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [97] htmlwidgets_1.6.4           S4Arrays_1.4.1             </span></span>
<span><span class="co">#&gt;  [99] uwot_0.2.2                  pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt; [101] gtable_0.3.5                modeltools_0.2-23          </span></span>
<span><span class="co">#&gt; [103] lmtest_0.9-40               SingleCellExperiment_1.26.0</span></span>
<span><span class="co">#&gt; [105] XVector_0.44.0              htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt; [107] dotCall64_1.1-1             bookdown_0.39              </span></span>
<span><span class="co">#&gt; [109] scales_1.3.0                Biobase_2.64.0             </span></span>
<span><span class="co">#&gt; [111] png_0.1-8                   harmony_1.2.0              </span></span>
<span><span class="co">#&gt; [113] knitr_1.47                  reshape2_1.4.4             </span></span>
<span><span class="co">#&gt; [115] nlme_3.1-164                cachem_1.1.0               </span></span>
<span><span class="co">#&gt; [117] zoo_1.8-12                  stringr_1.5.1              </span></span>
<span><span class="co">#&gt; [119] KernSmooth_2.23-24          libcoin_1.0-10             </span></span>
<span><span class="co">#&gt; [121] parallel_4.4.1              miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt; [123] desc_1.4.3                  pillar_1.9.0               </span></span>
<span><span class="co">#&gt; [125] grid_4.4.1                  vctrs_0.6.5                </span></span>
<span><span class="co">#&gt; [127] RANN_2.6.1                  promises_1.3.0             </span></span>
<span><span class="co">#&gt; [129] BiocSingular_1.20.0         beachmat_2.20.0            </span></span>
<span><span class="co">#&gt; [131] xtable_1.8-4                cluster_2.1.6              </span></span>
<span><span class="co">#&gt; [133] evaluate_0.24.0             mvtnorm_1.2-5              </span></span>
<span><span class="co">#&gt; [135] cli_3.6.3                   compiler_4.4.1             </span></span>
<span><span class="co">#&gt; [137] rlang_1.1.4                 crayon_1.5.3               </span></span>
<span><span class="co">#&gt; [139] future.apply_1.11.2         labeling_0.4.3             </span></span>
<span><span class="co">#&gt; [141] plyr_1.8.9                  fs_1.6.4                   </span></span>
<span><span class="co">#&gt; [143] stringi_1.8.4               viridisLite_0.4.2          </span></span>
<span><span class="co">#&gt; [145] deldir_2.0-4                BiocParallel_1.38.0        </span></span>
<span><span class="co">#&gt; [147] munsell_0.5.1               lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt; [149] spatstat.geom_3.2-9         Matrix_1.7-0               </span></span>
<span><span class="co">#&gt; [151] RcppHNSW_0.6.0              patchwork_1.2.0            </span></span>
<span><span class="co">#&gt; [153] sparseMatrixStats_1.16.0    future_1.33.2              </span></span>
<span><span class="co">#&gt; [155] shiny_1.8.1.1               SummarizedExperiment_1.34.0</span></span>
<span><span class="co">#&gt; [157] highr_0.11                  ROCR_1.0-11                </span></span>
<span><span class="co">#&gt; [159] igraph_2.0.3                memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [161] bslib_0.7.0</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Yue Lyu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
